internal/idler/user_idler.go

```diff
@@ -174,9 +176,14 @@ func (idler *UserIdler) doIdle() error {
 				reasonString = fmt.Sprintf("ActiveBuild BuildName:%s Last:%s", idler.user.ActiveBuild.Metadata.Name, idler.user.ActiveBuild.Status.StartTimestamp.Time)
 			}
 			idler.logger.WithField("attempt", fmt.Sprintf("(%d/%d)", idler.idleAttempts, idler.maxRetries)).Info("About to idle " + service + ", Reason: " + reasonString)
-			err := idler.openShiftClient.Idle(idler.openShiftAPI, idler.openShiftBearerToken, idler.user.Name+jenkinsNamespaceSuffix, service)
-			if err != nil {
-				return err
+
+			if idler.dryRun {
+				idler.logger.Info("Dry run is enabled, skipping idling.")
+			} else {
```

Comment:
how about just return here after logging to keep the nesting flat e.g. 

```go
if idler.dryRun {
   // log
   return nil
}

err := idler.openshiftClient.Idle(...) 

```

Reply:
sure, made the changes. I kept it like this because in case we'd want to do something in the function after the idling and we'd also want to cover that in dry run, but yeah, either works. PTAL